tests/Unit/DataStructures/Test_Mesh.cpp

```diff
@@ -76,14 +85,27 @@ SPECTRE_TEST_CASE("Unit.DataStructures.Mesh", "[DataStructures][Unit]") {
         mesh2d, {{2, 3}},
         {{Spectral::Basis::Legendre, Spectral::Basis::Legendre}},
         {{Spectral::Quadrature::Gauss, Spectral::Quadrature::GaussLobatto}});
-    const auto mesh2d_sliced_0 = mesh2d.slice_away(0);
-    test_extents_basis_and_quadrature(mesh2d_sliced_0, {{3}},
-                                      {{Spectral::Basis::Legendre}},
-                                      {{Spectral::Quadrature::GaussLobatto}});
-    const auto mesh2d_sliced_1 = mesh2d.slice_away(1);
-    test_extents_basis_and_quadrature(mesh2d_sliced_1, {{2}},
-                                      {{Spectral::Basis::Legendre}},
-                                      {{Spectral::Quadrature::Gauss}});
+    CHECK(mesh2d.slice_away(0) == Mesh<1>{3, Spectral::Basis::Legendre,
+                                          Spectral::Quadrature::GaussLobatto});
+    CHECK(mesh2d.slice_away(1) ==
+          Mesh<1>{2, Spectral::Basis::Legendre, Spectral::Quadrature::Gauss});
+    CHECK(mesh2d.slice_through() == Mesh<0>{});
+    CHECK(mesh2d.slice_through(0) == mesh2d.slice_away(1));
+    CHECK(mesh2d.slice_through(1) == mesh2d.slice_away(0));
+    CHECK(mesh2d.slice_through(0, 1) == mesh2d);
+    CHECK(mesh2d.slice_through(1, 0) ==
+          Mesh<2>{{{3, 2}},
+                  {{Spectral::Basis::Legendre, Spectral::Basis::Legendre}},
+                  {{Spectral::Quadrature::GaussLobatto,
+                    Spectral::Quadrature::Gauss}}});
+    CHECK(mesh2d.slice_through(0, 1, 1) ==
```

Comment:
This is a very strange operation.  Unless someone has a use case I think we should forbid it.

Reply:
Not sure, don't quite see why not just allow it. What's your take on reordering `mesh3d.slice(0,2,1)`?

Reply:
I dislike it because it isn't a slicing operation and I can't think of a use for it. 

I have no problem with the reordering operation.

Reply:
forbidding both would make it easier to implement :D

Reply:
Alright implemented it to allow reordering but no duplicate dimensions and `SliceDim<=Dim`.